src/core/server/saved_objects/migrations/model/extract_errors.ts

```diff
@@ -51,3 +51,32 @@ export function extractUnknownDocFailureReason(
     `'`
   );
 }
+
+/**
+ * Constructs migration failure message string for doc exceeds max batch size in bytes
+ */
+export const fatalReasonDocumentExceedsMaxBatchSizeBytes = ({
+  _id,
+  docSizeBytes,
+  maxBatchSizeBytes,
+}: {
+  _id: string;
+  docSizeBytes: number;
+  maxBatchSizeBytes: number;
+}) =>
+  `The document with _id "${_id}" is ${docSizeBytes} bytes which exceeds the configured maximum batch size of ${maxBatchSizeBytes} bytes. To proceed, please increase the 'migrations.maxBatchSizeBytes' Kibana configuration option and ensure that the Elasticsearch 'http.max_content_length' configuration option is set to an equal or larger value.`;
+
+/**
+ * Constructs migration failure message and logs message strings when an unsupported cluster routing allocation is configured.
+ * The full errorMessage is "[unsupported_cluster_routing_allocation] The elasticsearch cluster has cluster routing allocation incorrectly set for migrations to continue."
+ */
+export const fatalReasonClusterRoutingAllocationUnsupported = ({
```

Comment:
naming super-nit: It's a little hard to grok the purpose of these functions from their names. I might've gone with something like `formatClusterRoutingAllocationFailureReason`, `formatClusterRoutingAllocationFailureMessage`, and `formatDocumentExceedsMaxBatchSizeError`. But also naming's hard so 🤷